cpp/log/sqlite_db-inl.h

```diff
@@ -317,9 +328,27 @@ template <class Logged>
 int64_t SQLiteDB<Logged>::TreeSize() const {
   cert_trans::ScopedLatency latency(
       latency_by_op_ms.GetScopedLatency("tree_size"));
-  std::lock_guard<std::mutex> lock(lock_);
+  std::unique_lock<std::mutex> lock(lock_);
 
   CHECK_GE(tree_size_, 0);
+  sqlite::Statement statement(
```

Comment:
I'll measure the time to start and report back. I think the only place that is actually affected is `HttpHandler::ProxyInterceptor` (via `NodeIsStale`), so if there's any win to be had (and I'm hoping there is), I'll cache the staleness state in a separate PR first.